### PR TITLE
Login, Edit ,List Updated

### PR DESCRIPTION
**Pull request by Rajmohanrk**
update on the login guard , edit component and list component .

### CODE REVIEW

File: Maxis/wwwroot/login/shared/login.guard.ts [hunk 1]
```diff
@@ -7,12 +7,17 @@ export class LoginGuard implements CanActivate {
     constructor(private _router: Router) { }
 
     canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+        let adminPg = ['list', 'edit'];
         if (localStorage.getItem('currentUser')) {
             // logged in so return true
-            return true;
+            if (adminPg.indexOf(route.url[0].path.toString()) > -1 && localStorage.getItem('userrole') == "ADMIN") {
```
Comment (kamalakar1981) [inline Maxis/wwwroot/login/shared/login.guard.ts]: convert case and compare

File: Maxis/wwwroot/logout/logout.component.ts [hunk 2]
```diff
@@ -8,11 +8,10 @@ import { Router, ActivatedRoute } from '@angular/router';
 })
 
 export class LogoutComponent implements OnInit {
-    constructor(private _logoutService: LogoutService) {
+    constructor(private _logoutService: LogoutService) { }
+
+    ngOnInit() { }
```
Comment (kamalakar1981) [inline Maxis/wwwroot/logout/logout.component.ts]: remove

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 3]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
+import { NgForm } from '@angular/forms';
+
 @Component({
     selector: 'edit',
     templateUrl: 'wwwroot/user/edit/edit.component.html',
     styleUrls: ['wwwroot/user/edit/edit.component.css']
 })
+
 export class EditComponent implements OnInit, OnDestroy {
     @ViewChildren(FormControlName, { read: ElementRef }) formInputElements: ElementRef[];
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: use route and query param to navigate to edit page

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 4]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
 
     errorMessage: string;
     editform: FormGroup;
 
-    ulist: Userlist;
+    ulist: any;
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: use interface

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 5]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
     private sub: Subscription;
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: _

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 6]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
     displayMessage: { [key: string]: string } = {};
     private validationMessages: { [key: string]: { [key: string]: string } };
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: _

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 7]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
 
     constructor(private _editservice: EditService,
         private _formbuilder: FormBuilder,
+        private _fb: FormBuilder,
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: remove duplicate

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 8]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
+        //        range: 'Mobile No is required.'
+        //    },
+        //    Roles: {
+        //        range: 'user role is required.'
+        //    },
+        //    Department: {
+        //        range: 'Department is required.'
+        //    },
+        //    Title: {
+        //        range: 'title is required.'
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: remove commented code

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 9]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
+            Title: '',
+            Status: ['', <any>Validators.required]
         });
 
+
+        // subscribe to form changes  
+        this.subcribeToFormChanges();
+
+        // Update single value
+        (<FormControl>this.myForm.controls['name'])
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: remove commented code and clean unused code

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 10]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
+        const myFormStatusChanges$ = this.myForm.statusChanges;
+        const myFormValueChanges$ = this.myForm.valueChanges;
+
+        myFormStatusChanges$.subscribe(x => this.events.push({ event: 'STATUS_CHANGED', object: x }));
+        myFormValueChanges$.subscribe(x => this.events.push({ event: 'VALUE_CHANGED', object: x }));
+    }
 
+    save(model: any, isValid: boolean) {
+        this.submitted = true;
+        console.log(model, isValid);
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: remove console

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 11]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
+                this.onlistretrieved(ulist)
+            },
             (error: any) => this.errorMessage = <any>error
             );
     }
 
-    onlistretrieved(ulist: Userlist): void {
+    onlistretrieved(ulist: any): void {
         if (this.editform) {
             this.editform.reset();
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: not required

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 12]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
+        //    UserId: "this.ulist.userId",
+        //    Username: "this.ulist.userName",
+        //    Email: "this.ulist.emailId",
+        //    Mobile: "this.ulist.mobileNo",
+        //    Department: "this.ulist.department",
+        //    Title: "this.ulist.title"
         //});
+         //Update form model
+         
+        (<FormControl>this.myForm.controls['UserId'])
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: use route

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 13]
```diff
@@ -4,124 +4,214 @@ import { Subscription } from 'rxjs/Subscription';
...
+        (<FormControl>this.myForm.controls['Status'])
+            .setValue(ulist.Status, { onlySelf: true });
+         //TODO
+         //(<FormGroup>this.myForm)
+         //    .setValue(ulist, { onlySelf: true });
     }
 
     saveProduct(): void {
-        if (this.editform.dirty && this.editform.valid) {
+        if (this.myForm.dirty && this.myForm.valid) {
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: use meaningful name to form

File: Maxis/wwwroot/user/list/shared/list.service.ts [hunk 14]
```diff
@@ -20,7 +20,7 @@ export class ListService {
     }
     
     getUserlist(): Observable<Userlist[]> {
-        return this._http.get('USER/UsersList')
+        return this._http.get('User/UsersList')
             .map((response: Response) => <Userlist[]>response.json())
             .do(data => console.log('All: ' + JSON.stringify(data)))
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/list/shared/list.service.ts]: remove log

File: Maxis/systemjs.config.js [hunk 15]
```diff
@@ -49,17 +49,21 @@
...
+                main: './index.js',
                 defaultExtension: 'js'
             },
             'primeng': {
                 defaultExtension: 'js'
-            } ,
-            "ngx-dropdown": { "main": "index.js", "defaultExtension": "js" },
-            'ng2-bootstrap': {
+            },
+			'angular2-ui-switch': {
```
Comment (Rajmohanrk) [inline Maxis/systemjs.config.js]: code formated.

File: Maxis/package.json [hunk 16]
```diff
@@ -31,6 +33,7 @@
     "ng2-bootstrap": "^1.6.0",
     "ng2-select": "^1.2.0",
     "ng2-table": "^1.3.2",
+    "ngx-bootstrap": "^1.6.6",
```
Comment (Rajmohanrk) [inline Maxis/package.json]: removed the reference .

File: Maxis/wwwroot/user/edit/edit.component.ts [hunk 17]
```diff
@@ -4,124 +4,106 @@ import { Subscription } from 'rxjs/Subscription';
...
-        //    userName: this.ulist.userName,
-        //    emailId: this.ulist.emailId,
-        //    mobileNo: this.ulist.mobileNo,
-        //    department: this.ulist.department,
-        //    title: this.ulist.title,
-        //});
+        this.ulist = ulist;
+      
+        (<FormControl>this.editForm.controls['UserId'])
+            .setValue(ulist.UserId, { onlySelf: true });
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/edit/edit.component.ts]: Refactor this.

File: Maxis/wwwroot/user/list/list.component.html [hunk 18]
```diff
@@ -1,48 +1,174 @@
...
-<p-dataTable [value]="contacts">
-    <p-column *ngFor="let col of cols" [field]="col.field" [header]="col.header"></p-column>
-</p-dataTable>-->
+        <div class="page-wrapper-row">
+            <div class="page-wrapper-bottom">
+                <!-- BEGIN FOOTER -->
+                <!-- BEGIN INNER FOOTER -->
+                <div class="page-footer">
+                    <div class="container">
+                        2017 © OnNet All Rights Reserved
```
Comment (kamalakar1981) [inline Maxis/wwwroot/user/list/list.component.html]: make year dynamic

**Pull request closed without merging**